subsys/nrf_security/src/drivers/cracen/cracen_sw/include/cracen_sw_aes_ccm.h

```diff
@@ -0,0 +1,231 @@
+				      size_t ciphertext_size, size_t *ciphertext_length,
+				      uint8_t *tag, size_t tag_size, size_t *tag_length);
+
+/**
+ * @brief Verify and finish the software AES-CCM decryption operation.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ * @param[out] plaintext Output buffer for remaining plaintext (typically none for CCM).
+ * @param[in] plaintext_size Size of the plaintext buffer.
+ * @param[out] plaintext_length Pointer to store the actual plaintext length.
+ * @param[in] tag Authentication tag to verify.
+ * @param[in] tag_length Length of the authentication tag.
+ *
+ * @retval PSA_SUCCESS Authentication tag verified successfully.
+ * @retval PSA_ERROR_INVALID_SIGNATURE Authentication tag verification failed.
+ * @retval PSA_ERROR_BAD_STATE Operation not initialized.
+ */
+psa_status_t cracen_sw_aes_ccm_verify(cracen_aead_operation_t *operation, uint8_t *plaintext,
+				      size_t plaintext_size, size_t *plaintext_length,
+				      const uint8_t *tag, size_t tag_length);
+
+/**
+ * @brief Abort the software AES-CCM operation.
+ *
+ * Clears the operation context and any sensitive data.
+ *
+ * @param[in,out] operation Pointer to the AEAD operation structure.
+ *
+ * @retval PSA_SUCCESS Operation aborted successfully.
+ */
+psa_status_t cracen_sw_aes_ccm_abort(cracen_aead_operation_t *operation);
+
+/**
+ * @brief Perform a single-shot software AES-CCM encryption.
+ *
+ * @param[in] attributes Key attributes.
+ * @param[in] key_buffer Key material.
+ * @param[in] key_buffer_size Size of the key material.
+ * @param[in] alg Algorithm to use.
+ * @param[in] nonce Nonce value.
+ * @param[in] nonce_length Length of the nonce.
+ * @param[in] additional_data Additional authenticated data.
+ * @param[in] additional_data_length Length of the additional data.
+ * @param[in] plaintext Input plaintext to encrypt.
+ * @param[in] plaintext_length Length of the plaintext.
+ * @param[out] ciphertext Output buffer for ciphertext and tag.
+ * @param[in] ciphertext_size Size of the ciphertext buffer.
+ * @param[out] ciphertext_length Pointer to store the actual ciphertext and tag length.
+ *
+ * @retval PSA_SUCCESS Encryption completed successfully.
+ * @retval PSA_ERROR_BUFFER_TOO_SMALL Output buffer too small.
+ * @retval PSA_ERROR_INVALID_ARGUMENT Invalid nonce length or key size.
+ * @retval PSA_ERROR_NOT_SUPPORTED Unsupported algorithm or key type.
+ */
+psa_status_t cracen_sw_aes_ccm_encrypt(const psa_key_attributes_t *attributes,
+				       const uint8_t *key_buffer, size_t key_buffer_size,
+				       psa_algorithm_t alg, const uint8_t *nonce,
+				       size_t nonce_length, const uint8_t *additional_data,
+				       size_t additional_data_length, const uint8_t *plaintext,
+				       size_t plaintext_length, uint8_t *ciphertext,
+				       size_t ciphertext_size, size_t *ciphertext_length);
+
+/**
+ * @brief Perform a single-shot software AES-CCM decryption.
```

Comment:
```suggestion
 * @brief Perform a single software AES-CCM decryption.
```
To avoid the colloquialism.